packages/realm/src/app-services/App.ts

```diff
@@ -35,6 +35,70 @@ import {
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type AnyApp = App<any, any>;
 
+/**
+ * This describes the persistence modes for metadata.
+ */
+export enum MetadataMode {
+  /**
+   * Persist {@link User} objects, but do not encrypt them.
+   */
+  NoEncryption = "NoEncryption",
+  /**
+   * Persist {@link User} objects in an encrypted store.
+   */
+  Encryption = "Encryption",
+  /**
+   * Do not persist {@link User} objects.
+   */
+  NoMetadata = "NoMetadata",
+}
+
+/**
+ * This describes how the app's metadata is handled.
+ */
+export type MetadataType = {
+  /**
+   * The different modes for storing meta data in Realm Apps.
+   * @since 12.2.0
+   */
+  mode: MetadataMode;
+  /**
+   * The 512-bit (64-byte) encryption key used to encrypt and decrypt meta data in Realm Apps.
+   * This will not change the encryption key for individual Realms. This should still be set in
+   * {@link Configuration.encryptionKey} when opening the Realm.
+   * @since 12.2.0
+   */
+  encryptionKey?: ArrayBuffer;
+};
+
+/** internal */
+function toBindingMetadataMode(arg: MetadataMode): binding.MetadataMode {
+  const bindingMetadataMode = inverseTranslationTable[arg];
+  assert(bindingMetadataMode !== undefined, `Unexpected metadata mode: ${arg}`);
+  return bindingMetadataMode;
+}
+
+/** @internal */
+export function fromBindingMetadataModeToNumericMetadataMode(arg: binding.MetadataMode): MetadataMode {
+  // For now, these map 1-to-1
+  return arg as unknown as MetadataMode;
+}
```

Comment:
Don't think I've seen this function being called anywhere?

Reply:
No, I added it for completeness and it can safely be removed.